contribute-docs/api-docs/kibana-api-docs-quickstart.md

```diff
@@ -0,0 +1,307 @@
+Edit the TypeScript route definitions in your plugin code. Add JSDoc comments, request/response schemas, and examples as needed, per the [checklist](checklist.md).
+
+:::{note}
+**CI will automatically regenerate the OpenAPI files when you push your `.ts` changes.** The next two steps show how to capture the snapshot and add examples locally, which is useful for validating changes before pushing or debugging issues.
+:::
+
+::::
+
+::::{tab-item} Manual YAML
+:sync: manual
+Edit the YAML files directly in the `oas_docs` directory. Refer to the README alongside each file for specific guidance on adding summaries, descriptions, tags, metadata, links, and examples.
+
+Review the the [checklist](checklist.md) for best practices.
+
+Once you've made your changes, skip the next two steps and proceed to "Generate docs".
+::::
+
+::::::
+
+::::::{step} Add examples to your routes
+
+Beyond schema definitions, providing concrete request and response examples significantly improves API documentation usability.
+
+::::{dropdown} Inline TypeScript examples
+
+For code-generated APIs, you can add examples directly in your route definitions. Examples are type-checked at development time, so shape errors are caught during authoring.
+
+```typescript
+.addVersion({
+  version: '2023-10-31',
+  options: {
+    oasOperationObject: () => ({
+      requestBody: {
+        content: {
+          'application/json': {
+            examples: {
+              fooExample1: {
+                summary: 'An example foo request',
+                value: {
+                  name: 'Cool foo!',
+                } as FooResource,
+              },
+            },
+          },
+        },
+      },
+    }),
+  },
+  // ...
+})
+```
+::::
+::::{dropdown} YAML examples
+
+**For code-generated APIs**, reference a YAML file in your route definition:
+
+```typescript
+import path from 'node:path';
+
+const oasOperationObject = () => path.join(__dirname, 'foo.examples.yaml');
+
+.addVersion({
+  version: '2023-10-31',
+  options: {
+    oasOperationObject,
+  },
+  validate: {
+    request: {
+      body: fooResource,
+    },
+    response: {
+      200: {
+        body: fooResourceResponse,
+      },
+    },
+  },
+})
+```
+
+**For Manual YAML**, add examples directly in your YAML files in the `oas_docs` directory.
+
+**Example YAML structure:**
+
+```yaml
+requestBody:
+  content:
+    application/json:
+      examples: # Use examples (plural), not example (deprecated)
+        fooExample:
+          summary: Foo example
+          description: An example request of creating foo
+          value:
+            name: 'Cool foo!'
+            enabled: true
+responses:
+  200:
+    content:
+      application/json:
+        examples:
+          exampleResponse:
+            summary: Successful response
+            value:
+              id: '12345'
+              name: 'Cool foo!'
+              enabled: true
+```
+::::
+
+::::::
+
+::::::{step} (optional) Capture OAS snapshot
```

Comment:
I think "OAS" isn't a widely-used acronym, so something like this might be clearer:
```suggestion
::::::{step} (optional) Capture code-generated output
```